diff --git a/react-native-udp.podspec b/react-native-udp.podspec
index 7450cc7d0862aadfb47d796929c801a3dc423a57..fa3e42c0152ef2d87536b8c2e484f64d525e35ec 100644
--- a/react-native-udp.podspec
+++ b/react-native-udp.podspec
@@ -9,7 +9,8 @@ Pod::Spec.new do |s|
   s.homepage       = package_json["homepage"]
   s.license        = package_json["license"]
   s.author         = { package_json["author"] => package_json["author"] }
-  s.platform       = :ios, "7.0"
+  s.ios.deployment_target = "7.0"
+  s.tvos.deployment_target = "15.1"
   s.source         = { :git => package_json["repository"]["url"], :tag => "v#{s.version}" }
   s.source_files   = 'ios/**/*.{h,m}'
   s.dependency 'React-Core'
