
diff --git a/node_modules/react-native-uitextview/ios/RNUITextView.mm b/node_modules/react-native-uitextview/ios/RNUITextView.mm
index a77f196..c928743 100644
--- a/node_modules/react-native-uitextview/ios/RNUITextView.mm
+++ b/node_modules/react-native-uitextview/ios/RNUITextView.mm
@@ -4,10 +4,11 @@
 
 #import <React/RCTConversions.h>
 
-#import <react-native-uitextview/EventEmitters.h>
-#import <react-native-uitextview/Props.h>
-#import <react-native-uitextview/RCTComponentViewHelpers.h>
-#import <react/renderer/textlayoutmanager/RCTAttributedTextUtils.h>
+#import <EventEmitters.h>
+#import <Props.h>
+#import <RCTComponentViewHelpers.h>
+//#import "./generated/RNUITextViewSpec/R"
+#import "React_FabricComponents/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTAttributedTextUtils.h"
 
 #import "RCTFabricComponentsPlugins.h"
 
diff --git a/node_modules/react-native-uitextview/ios/RNUITextViewChild.mm b/node_modules/react-native-uitextview/ios/RNUITextViewChild.mm
index 3eebf75..ec4658f 100644
--- a/node_modules/react-native-uitextview/ios/RNUITextViewChild.mm
+++ b/node_modules/react-native-uitextview/ios/RNUITextViewChild.mm
@@ -3,9 +3,9 @@
 #import "RNUITextView.h"
 #import "RNUITextViewChildComponentDescriptor.h"
 
-#import <react-native-uitextview/EventEmitters.h>
-#import <react-native-uitextview/Props.h>
-#import <react-native-uitextview/RCTComponentViewHelpers.h>
+#import "./generated/RNUITextViewSpec/EventEmitters.h"
+#import "./generated/RNUITextViewSpec/Props.h"
+#import <RCTComponentViewHelpers.h>
 
 #import "RCTFabricComponentsPlugins.h"
 #import "Utils.h"
diff --git a/node_modules/react-native-uitextview/ios/RNUITextViewChildShadowNode.h b/node_modules/react-native-uitextview/ios/RNUITextViewChildShadowNode.h
index 838a8b4..d74d9e2 100644
--- a/node_modules/react-native-uitextview/ios/RNUITextViewChildShadowNode.h
+++ b/node_modules/react-native-uitextview/ios/RNUITextViewChildShadowNode.h
@@ -1,8 +1,8 @@
 #pragma once
 
-#include <react-native-uitextview/EventEmitters.h>
-#include <react-native-uitextview/Props.h>
-#include <react-native-uitextview/States.h>
+#include <EventEmitters.h>
+#include <Props.h>
+#include <States.h>
 #include <react/renderer/components/view/ConcreteViewShadowNode.h>
 
 namespace facebook::react {
diff --git a/node_modules/react-native-uitextview/ios/RNUITextViewShadowNode.h b/node_modules/react-native-uitextview/ios/RNUITextViewShadowNode.h
index 23da835..2e76ff6 100644
--- a/node_modules/react-native-uitextview/ios/RNUITextViewShadowNode.h
+++ b/node_modules/react-native-uitextview/ios/RNUITextViewShadowNode.h
@@ -1,11 +1,14 @@
 #pragma once
 
-#include <react-native-uitextview/EventEmitters.h>
-#include <react-native-uitextview/Props.h>
+#include "./generated/RNUITextViewSpec/EventEmitters.h"
+#include "./generated/RNUITextViewSpec/Props.h"
+
 #include <react/renderer/components/view/ConcreteViewShadowNode.h>
-#include <react/renderer/textlayoutmanager/TextLayoutManager.h>
 #include <react/renderer/core/LayoutContext.h>
 #include <react/renderer/core/ShadowNode.h>
+#include "React_FabricComponents/react/renderer/textlayoutmanager/TextLayoutManager.h"
+
+
 
 namespace facebook::react {
 
