
diff --git a/node_modules/react-native/ReactCommon/react/renderer/textlayoutmanager/TextLayoutManager.h b/node_modules/react-native/ReactCommon/react/renderer/textlayoutmanager/TextLayoutManager.h
index 42f22de..cf80a95 100644
--- a/node_modules/react-native/ReactCommon/react/renderer/textlayoutmanager/TextLayoutManager.h
+++ b/node_modules/react-native/ReactCommon/react/renderer/textlayoutmanager/TextLayoutManager.h
@@ -10,8 +10,8 @@
 #include <react/renderer/attributedstring/AttributedStringBox.h>
 #include <react/renderer/attributedstring/ParagraphAttributes.h>
 #include <react/renderer/core/LayoutConstraints.h>
-#include <react/renderer/textlayoutmanager/TextLayoutContext.h>
-#include <react/renderer/textlayoutmanager/TextMeasureCache.h>
+#include "./TextLayoutContext.h"
+#include "./TextMeasureCache.h"
 #include <react/utils/ContextContainer.h>
 #include <memory>
 